src/com/lftechnology/java/training/srijan/basics/coreapp/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,390 @@
+				prepareStatementInsertEmployeeDetails.setString(
+					2, e.getFullName());
+				prepareStatementInsertEmployeeDetails.setString(
+					3, e.getDepartment());
+				prepareStatementInsertEmployeeDetails.setString(
+					4, e.getAddress());
+				prepareStatementInsertEmployeeDetails.setString(
+					5, e.getUserRole().name());
+				prepareStatementInsertEmployeeDetails.setBoolean(
+					6, e.getIsTerminated());
+				prepareStatementInsertEmployeeDetails.setString(
+					7, e.getPassword());
+				prepareStatementInsertEmployeeDetails.executeUpdate();
+			}
+		}
+		catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception{0}", ex);
+		}
+		finally {
+			try {
+				prepareStatementInsertEmployeeDetails.close();
+				con.close();
+			}
+			catch (SQLException e1) {
+				LOGGER.log(Level.INFO, "SQL Exception{0}", e1);
+			}
+		}
+	}
+
+	@Override
+	public Employee findById(int employeeIdToUpdate) {
+
+		Connection con = null;
+		ResultSet resultSet = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareStatementUpdateInfo = null;
+		Employee employee = new Employee();
+		try {
+			if (con != null) {
+				String query = "Select * from employeeDetails where id = ?";
+				prepareStatementUpdateInfo = con.prepareStatement(query);
+				prepareStatementUpdateInfo.setInt(1, employeeIdToUpdate);
+				resultSet = prepareStatementUpdateInfo.executeQuery();
+				while (resultSet.next()) {
+					employee.setId(resultSet.getInt("id"));
+					employee.setUserName(resultSet.getString("username"));
+					employee.setFullName(resultSet.getString("fullName"));
+					employee.setDepartment(resultSet.getString("department"));
+					employee.setAddress(resultSet.getString("address"));
+					employee.setIsTerminated(
+						resultSet.getBoolean("is_terminated"));
+				}
+			}
+		}
+		catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception:{0}", ex);
+		}
+		finally {
+			try {
+				resultSet.close();
+				prepareStatementUpdateInfo.close();
+				con.close();
+			}
+			catch (SQLException ex) {
+				LOGGER.log(Level.INFO, "SQL Exception {0}", ex);
+			}
+		}
+		return employee;
+	}
+
+	/**
+	 * <p>Checks login Validation depending on userName and password</p>
+	 * 
+	 * @author srijan
+	 */
+	@Override
+	public UserRole loginValidation(Employee e) {
+
+		ResultSet resultSet = null;
+		Connection con = null;
+		con = DbConnection.dbConnection();
+		PreparedStatement prepareStatementLoginCheck = null;
+		try {
+			if (con != null) {
+				String query =
+					"Select * from employeeDetails where username = ? and password = ? and is_terminated = ?";
+				prepareStatementLoginCheck = con.prepareStatement(query);
+				prepareStatementLoginCheck.setString(1, e.getUserName());
+				prepareStatementLoginCheck.setString(2, e.getPassword());
+				prepareStatementLoginCheck.setBoolean(3, false);
+				resultSet = prepareStatementLoginCheck.executeQuery();
+				if (resultSet.next()) {
+					LOGGER.log(Level.INFO, "Login Successful");
+					UserRole role =
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "role". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)